packages/web-app-files/src/components/SideBar/Shares/Links/CreateQuickLink.vue

```diff
@@ -31,36 +43,36 @@ export default defineComponent({
     }
   },
   emits: ['createPublicLink'],
-  setup() {
+  setup(props, { emit }) {
     const { can } = useAbility()
+    const { $gettext } = useGettext()
+
     const canCreatePublicLinks = computed(() => can('create-all', 'PublicLink'))
-    return { canCreatePublicLinks }
-  },
-  computed: {
-    heading() {
-      return this.$gettext('Quick link')
-    },
-    createLinkHint() {
-      return this.$gettext('Create quick link')
-    },
-    createLinkLabel() {
-      return this.$gettext('Create link')
-    },
-    noLinkLabel() {
-      return this.$gettext('No link')
-    }
-  },
-  methods: {
-    createQuickLink() {
-      this.$emit('createPublicLink', {
+    const resource = inject<Resource>('resource')
+    const capabilities = inject<Capabilities>('capabilities')
+    const createQuickLink = () => {
+      const allowResharing = capabilities.capabilities.files_sharing?.resharing
+      const emitData = {
         link: {
-          name: this.$gettext('Quicklink'),
-          permissions: this.canCreatePublicLinks ? 1 : 0,
-          expiration: this.expirationDate.enforced ? this.expirationDate.default : null,
+          name: $gettext('Quicklink'),
+          permissions: LinkShareRoles.getByName('none', resource.isFolder).bitmask(allowResharing),
+          expiration: props.expirationDate.enforced ? props.expirationDate.default : null,
           quicklink: true,
           password: false
         }
-      })
+      }
+      if (!canCreatePublicLinks.value) {
+        emit('createPublicLink', emitData)
+      }
+      const capabilitiesRoleName = capabilities.capabilities.files_sharing?.quickLink?.default_role
```

Comment:
Good candidate for a `useCapabilityFilesSharingQuickLinkDefaultRole` 🙂  You can then set `viewer` is default.